job_boards_scrapers/linkedin.py

```diff
@@ -0,0 +1,68 @@
+import re
+from http.client import InvalidURL
+from typing import Optional
+from typing import Union
+
+import requests
+
+from job_boards_scrapers.base import JobBoard
+from job_boards_scrapers.base import JobBoardType
+from job_boards_scrapers.base import JobInfo
+
+
+class LinkedIn(JobBoard):
+    """
+    Represents a LinkedIn job board
+    """
+
+    _VIEW_LINK_PREFIX = "https://www.linkedin.com/jobs/view"
+    _RECOMMENDED_LINK_REGEX = re.compile(r".*\?currentJobId=[0-9]*$")
+
+    def __init__(self) -> None:
+        self._job_board_name = JobBoardType.LINKEDIN
+        self._title_key = "top-card-layout__title"
+        self._company_key = "topcard__org-name-link"
+        self._company_pic_key = "sub-nav-cta__image"
+        self._img_src_key = "data-delayed-url"
+        self._time_ago_key = "posted-time-ago__text"
+        self._description_key = "description__text"
+        self._location_key = "sub-nav-cta__meta-text"
+
+    def get_job_info(self, job: Union[str, int]) -> Optional[JobInfo]:
+        is_direct_view = False
+
+        if isinstance(job, int):
+            job = self._url_from_job_id(job)
+            is_direct_view = True
+
+        if requests.head(job).status_code >= 400:
+            raise InvalidURL()
```

Comment:
Right now, I just temporarily raise an error. But this is dangerous since this is not a serverside error but rather the input error. So we actually want to inform the user instead. Will follow up with a different PR.

Reply:
Acknowledged.